Install drush via PEAR channel

dkingofpa opened the pull request:
- drush is installed via PEAR channel.
- added recipe to install drush_make.
- added a README file.
- added attributes files for recipe defaults.
- updated metadata to include all pieces of the cookbook.

Commit:
Updates cookbook so drush is installed via PEAR channel.

- drush is installed via PEAR channel.
- added recipe to install drush_make.
- added a README file.
- added attributes files for recipe defaults.
- updated metadate to include all pieces of the cookbook.

msonnabaum commented on attributes/default.rb:
```diff
@@ -0,0 +1,18 @@
+# limitations under the License.
+#
+
+default[:drush][:version] = "4.5.0"
```
Not sure if this is needed. I think I'd rather let pear always default to the latest stable release in the default recipe, and then just have recipes for running 5.x-dev and 4.x-dev.

dkingofpa replied:
Pear search doesn't currently work with the preferred state.  So we have to specify a version for now.  Unless I completely missed something.  https://pear.php.net/bugs/bug.php?id=19138

msonnabaum replied:
Yeah, that's probably true. This is fine then for now.

dkingofpa replied:
And yes, I think it would be nice to have a "default[:drush][:preferred_state]" attribute in the future.

dkingofpa replied:
I added a note about why there's no preferred_state attribute.

msonnabaum commented on recipes/default.rb:
```diff
@@ -1,5 +1,5 @@
-# Author:: Mark Sonnabaum <mark.sonnabaum@acquia.com>
```
Maybe let's not remove this line? :)

dkingofpa replied:
I removed it since I didn't see an Author line in any of the opscode recipes.  When generating a cookbook using "knife cookbook create", the stub recipe file doesn't include an Author line.  I figured the maintainer_email line in the metadata.rb was enough.  And I didn't add an Author line to the make recipe.  But if you don't want to remove that line, no problem.

msonnabaum replied:
If you look through the opscode cookbooks there are actually many instances of this. It might not appear in the stub because cookbook_copyright is not set? Not sure, see this: http://www.opscode.com/blog/2011/09/07/guide-to-writing-chef-cookbooks/

Anyhow, I think it's important since these recipes get traded around a lot and it's useful to know where it originated from. Feel free to add an additional line for yourself if you'd like.

dkingofpa replied:
Yup, I didn't have those cookbook boilerplate settings configured.  http://wiki.opscode.com/display/chef/Knife#Knife-Knifeconfiguration

How would I add my author lines to the make recipe and attribute files at this point?  Another pull request?

msonnabaum replied:
I think if you commit the change to your branch it'll show up here.

dkingofpa replied:
Added your author line back and added some on the new files.

msonnabaum commented on recipes/default.rb:
```diff
@@ -15,26 +15,23 @@
-    only_if { File.exists?("/usr/share/drush/drush") }
-  end
+# Upgrade PEAR to latest stable version
+# For example, 1.9.4 works for drush, but 1.9.0 does not
```
I'd rather be more explicit about when to upgrade pear. See how I handle this in my phpunit cookbook: https://github.com/msonnabaum/chef-phpunit/blob/master/recipes/default.rb#L26

dkingofpa replied:
Ah, cool...I like your solution better.  Yeah, I didn't know exactly what version of pear was needed for drush to work, hence the ambiguity.  :)  It's frustrating that opscode won't update pear via the php cookbook.  http://tickets.opscode.com/browse/COOK-747

dkingofpa replied:
I pulled in your pear upgrade block from phpunit instead.

Commit:
Fix spelling

Commit:
Fix attributes name in comment and add pear note

- Changed attributes name in comment at the top from drush to default.
- Added note about pear search bug regarding preferred state.

Commit:
Add author line and fix for some formatting

Commit:
Better formatting for metadata recipe descriptions

Commit:
Add author line to make.rb recipe

Commit:
Fix recipe formatting and names in comments

- Removed extra space between Cookbook name label and the actual name.
- Changed the recipe names to the correct values.

Commit:
Add more explicit upgrade check for pear

- Copied Mark's pear upgrade block from phpunit cookbook. Now it will
  only upgrade pear if the current version is less than 1.9.1

Commit:
Add author line back in to comments

dkingofpa commented:
I updated my repo with the changes you mentioned and added a few fixes I found.

Commit:
Install Console_Table package via pear

Drush requires Console_Table to run. Download it by default so we
do not have to run drush with sudo just to download the package.
Otherwise, we may end up with the following error:

vagrant@lucid32:~$ drush status
The drush command 'status' could not be found.                       [error]
Drush needs a copy of the PEAR Console_Table library in order to     [error]
function, and the attempt to download this file automatically failed
because you do not have permission to write files in
/usr/share/php/drush/includes. To continue you will need to download
the 1.1.3 package from http://pear.php.net/package/Console_Table,
extract it, and copy the Table.php file into Drush's directory as
/usr/share/php/drush/includes/table.inc.

msonnabaum commented:
Ok, this is fine. Merged.

frazras commented:
Im getting a bug with this:

STDERR: Drush needs to download a library from                                   [error]
http://download.pear.php.net/package/Console_Table-1.1.3.tgz in order
to function, and the attempt to download this file automatically
failed because you do not have permission to write to the library
directory /usr/share/php/drush/lib. To continue you will need to
manually download the package from
http://download.pear.php.net/package/Console_Table-1.1.3.tgz, extract
it, and copy the directory into your /usr/share/php/drush/lib
directory.
